config.go

```diff
@@ -269,25 +270,30 @@ func loadConfig() (*config, error) {
 	loadConfigError := func(err error) (*config, error) {
 		return nil, err
 	}
-
 	// Default config.
 	cfg := defaultConfig
-
+	// load environment variables into the config overriding the default config
+	err := env.Parse(&cfg)
+	if err != nil {
+		fmt.Printf("%+v\n", err)
```

Comment:
`return loadConfigError(err)`

Your bad value test is a good example of why not to let this slide by with a console message.